gtfu/src/main/java/gtfu/Time.java

```diff
@@ -197,4 +198,19 @@ public static String formatDate(String format, Date date) {
     public static long getNextDaySeconds(long timestamp) {
         return timestamp + SECONDS_PER_DAY;
     }
-}
\ No newline at end of file
+
+    public static Date getDateFromYYYYMMDD(String yyyyMMDD) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMdd");
+        try {
+            return dateFormat.parse(yyyyMMDD);
+        } catch (ParseException e) {
+            throw new RuntimeException("Error parsing date: " + e.getMessage(), e);
+        }
+    }
+
+    public static int getDaysUntilDateLong(String yyyyMMDD) {
```

Comment:
It maybe more versatile to have something like `getDayDifference(String yyyyMMDD)`, which doesn't make assumptions about whether the date passed in is before or after the current date. For extra points you can pass two dates into the method, and then have a convenience method that only takes one arg, and sets the second one to `today`. Method returns negative number if arg 1 is less than arg2, i.e. target date is before reference date, positive number otherwise.

Reply:
updated! I ended up doing another "getDaysUntilDate" function, but one that utilizes a "differenceBetweenDates" function. The resulting logic is a little funny, as types get converted back and forth and back, but I think result is something that is intuitive to interact with.